### PR TITLE
 Removed @SafeVarargs annotation since it's not available until API 19

### LINKED ISSUE #5463: NoClassDefFoundError (SafeVarargs) API 17
#### Goal

> Use realm-java on android api level 17

#### Expected Results

> realm-java does not crash on api level 17

#### Actual Results

>  I am encountering the following exception when I upgrade my project from 3.7.2 to 4.1.0: java.lang.NoClassDefFoundError: java/lang/SafeVarargs. This is only occurring on api level 17, all apis above 17 seem to work fine. Before this crash I get an error specifically regarding RealmList: 
E/dalvikvm: Unable to resolve Lio/realm/RealmList; annotation class 7880

#### Steps & Code to Reproduce

> Fairly certain this is occurring due to the recent use of the SafeVarargs annotation in RealmList and RealmConfiguration. This can be reproduced by using a RealmList within a project on the api level 17 emulator.

#### Version of Realm and tooling
Realm version(s): 4.1.0

Realm sync feature enabled: no

Android Studio version: 2.3.3

Which Android version and device: Emulator, API 17.

### PR DESCRIPTION
**Pull request by nhachicha**
fixes #5463

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 ### Bug Fixes
 
 * Fixed the compile warnings of using deprecated method `RealmProxyMediator.getTableName()` in generated mediator classes (#5455).
+* [ObjectServer] now retrying network query when encountering any `IOException` (#5453).
+* Fixed a `NoClassDefFoundError` due to using `@SafeVarargs` (#5463).
```
Comment (Zhuinden) [inline CHANGELOG.md]: below API 19?

Comment (nhachicha) [reply CHANGELOG.md]: yep

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@
 ### Bug Fixes
 
 * Fixed the compile warnings of using deprecated method `RealmProxyMediator.getTableName()` in generated mediator classes (#5455).
+* [ObjectServer] now retrying network query when encountering any `IOException` (#5453).
+* Fixed a `NoClassDefFoundError` due to using `@SafeVarargs` (#5463). 
 
 ### Interal
```
Comment (Zhuinden) [inline CHANGELOG.md]: Missing `n`

Comment (nhachicha) [reply CHANGELOG.md]: are you referring to "a `NoClassDefFoundError`"? I think `a` is more appropriate in this case https://www.englishclub.com/pronunciation/a-an.htm 🤓

Comment (Zhuinden) [reply CHANGELOG.md]: No I mean `Interal` :D

Comment (nhachicha) [reply CHANGELOG.md]: Ah yeah! let's blame @cmelchior

**Comment by nhachicha** [comment]
@realm/java

File: CHANGELOG.md (hunk 2 above)
Comment (cmelchior) [reply CHANGELOG.md]: Lets not  😛

**Pull request merged**